crates/oxc_linter/src/rules/eslint/no_octal_escape.rs

```diff
@@ -0,0 +1,172 @@
+use oxc_ast::AstKind;
+use oxc_diagnostics::OxcDiagnostic;
+use oxc_macros::declare_oxc_lint;
+use oxc_span::Span;
+use regex::Regex;
+use std::sync::OnceLock;
+
+use crate::{context::LintContext, rule::Rule, AstNode};
+
+fn no_octal_escape_diagnostic(span: Span, sequence: &str) -> OxcDiagnostic {
+    OxcDiagnostic::warn(format!("Don't use octal: '\\{sequence}'. Use '\\u....' instead."))
+        .with_label(span)
+}
+
+#[derive(Debug, Default, Clone)]
+pub struct NoOctalEscape;
+
+declare_oxc_lint!(
+    /// ### What it does
+    ///
+    /// Disallows the use of octal escape sequences in string literals.
+    ///
+    /// ### Why is this bad?
+    ///
+    /// Octal escape sequences are deprecated and can lead to unexpected behavior. Modern code should use Unicode or hexadecimal escape sequences instead.
+    ///
+    /// ### Examples
+    ///
+    /// Examples of **incorrect** code for this rule:
+    /// ```js
+    /// const str = "\251";
+    /// const str = "\1";
+    /// ```
+    ///
+    /// Examples of **correct** code for this rule:
+    /// ```js
+    /// const str = "\u00A9";
+    /// const str = "\xA9";
+    /// ```
+    NoOctalEscape,
+    correctness,
+);
+
+impl Rule for NoOctalEscape {
+    fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
+        if let AstKind::StringLiteral(literal) = node.kind() {
+            if let Some(raw) = &literal.raw {
+                if let Some(captures) = get_octal_escape_pattern().captures(raw) {
+                    if let Some(sequence) = captures.get(1) {
+                        ctx.diagnostic(no_octal_escape_diagnostic(literal.span, sequence.as_str()));
+                    }
+                }
+            }
+        }
+    }
+}
+
+static OCTAL_ESCAPE_PATTERN: OnceLock<Regex> = OnceLock::new();
+
+fn get_octal_escape_pattern() -> &'static Regex {
+    OCTAL_ESCAPE_PATTERN.get_or_init(|| {
+        Regex::new(r"^(?:[^\\]|\\.)*?\\([0-3][0-7]{1,2}|[4-7][0-7]|(08|09)|[1-7])").unwrap()
+    })
```

Comment:
you can use a lazy_static

https://github.com/oxc-project/oxc/blob/bd0693bcd01cc826a4717779558e06ce99262732/crates/oxc_linter/src/rules/unicorn/throw_new_error.rs#L96-L99

Reply:
I learned this. fixed: https://github.com/oxc-project/oxc/pull/8151/commits/b30306410945a7ab4932a5862ec0387c7e0cd29f